### DISCUSSION TITLE
New qtransaltext with slugs kill all old slugs.  _qts_slug  vs  qtranslate_slug

### DISCUSSION THREAD

**Discussion by andreysneg** [qtranslate/qtranslate-xt#1205] [General]
Very bad idea change metafield   _qts_slug_ to  qtranslate_slug_

Whyyyy ?

What is problem with _qts_slug  ? 

Its kill all compatibility for site what use qts_slug  plugin . Posts, products, pages , all  slugs is dead with new plugin.  I must change slug for 10.000 products ? at 100 sites ? why ?

All old sits slugs is dead with new qtranslatext plugin and slug activated.

Now i dont understand work old _qts_slugs or not. i see what work at  frontend...  but qts slugs is empty at backend.

I cant use new slugs addon for old sites . 

Absolutly dont understand idea for change  to  qtranslate_slug_  

Please change to standart _qts_slug  or make options for old sites.  Thank you.

**Comment by herrvigg** [comment]
This has been widely documented, please read carefully the upgrade notices, changelogs and README. You have nothing to do but follow the instructions for migration. It should be straightforward in two clicks.
https://github.com/qtranslate/qtranslate-xt/blob/master/modules/slugs/README.md#ive-been-using-the-qts-plugin-how-can-i-find-the-existing-slugs
Migration PR of QTS to Slugs module: https://github.com/qtranslate/qtranslate-xt/pull/1171

The reason has been explained, QTS is a plugin for qTranslate-X (before XT) and there are critical issues related to the uninstall of that plugin. It's part of the discussion https://github.com/qtranslate/qtranslate-xt/discussions/1106 you initiated yourself. More generally it's about ownership of the data. Now that the QTS plugin has become a module in QT-XT the ownership becomes clearer by separating the module from the plugin. This has been released for a while so we won't/can't go back on this choice now.

**Reply by leicestersq** [reply]
Slug integration is working flawlessly! We have a site with about 1000 urls created with the old slug translation plugin which were perfectly imported by Q-Translate-xt with just one click!! Bravo for all the good work keeping this plugin alive and current in 2022.

**Reply by herrvigg** [reply]
Many thanks!

**Reply by herrvigg** [reply]
@andreysneg Did the migration work correctly?

**Comment by andreysneg** [comment]
I have 20 dynamic websites what use plugins for import-export products  . Plugins use old  _qts  fields in PHP.  Change plugins - is very hard and long work.
All 20 websites cant use new plugins Qtranslate XT for work.  

I still don’t see any reason to change the old _qts  names, but I understand that you won’t return anything back.
You didn't think of other external plugins that might use these _qts  fields.

Topic can be closed.

**Reply by herrvigg** [reply]
True, we didn't think much of other plugins but qTranslate-X is abandoned and so is the old QTS integration plugin. And this was creating other problems for example with the uninstallation. So we integrated QTS fully in QTX differently. I hope you can upgrade those plugins on their side.
Until then, you could delay the migration to later, fork QTX and do these temporary changes.
- For the META PREFIX in principle it's enough to change the value here :
https://github.com/qtranslate/qtranslate-xt/blob/6ed2efa4202b6d5fe25f4e708018dbd3d5d34ec1/modules/slugs/slugs.php#L2
Set QTX_SLUGS_META_PREFIX = QTX_SLUGS_LEGACY_QTS_META_PREFIX (`_qts_slug_`).
- For the options it depends how the other plugins work, but you could set QTX_OPTIONS_MODULE_SLUGS = QTX_SLUGS_LEGACY_QTS_OPTIONS_NAME  (`qts_options`). But we also removed the prefix QTX_SLUGS_LEGACY_QTS_OPTIONS_PREFIX (`_qts_`) from the keys so you'd also need to change a bit of code.
